File: src/types/Stock/DetalleStock.ts
import { Ingrediente } from "../Ingredientes/Ingrediente";
import { Medida } from "../Ingredientes/Medida";
import { ArticuloVenta } from "../Productos/ArticuloVenta";

export class DetalleStock {
    id: number = 0;
    cantidad: number = 0;
    costoUnitario: number = 0;
    medida: Medida = new Medida();
    ingrediente: Ingrediente = new Ingrediente();
    articuloVenta: ArticuloVenta = new ArticuloVenta();
    borrado: string = '';
    modificarPrecio: boolean = false;

    constructor() {

    }
}

File: server/api/src/main/java/main/entities/Stock/DetalleStock.java
package main.entities.Stock;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import jakarta.persistence.*;
import lombok.*;
import main.entities.Ingredientes.Ingrediente;
import main.entities.Ingredientes.Medida;
import main.entities.Productos.ArticuloVenta;

@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@Entity
@Builder
@ToString
@Table(name = "detalle_stock", schema = "buen_sabor")
public class DetalleStock {
    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private Long id;
    @Column(name = "cantidad")
    private double cantidad;
    @JsonIgnoreProperties(value = {"sucursales"}, allowSetters = true)
    @ManyToOne
    @JoinColumn(name = "id_medida")
    private Medida medida;
    @Column(name = "costo_unitario")
    private double costoUnitario;
    @Column(name = "subtotal")
    private double subTotal;

    @ManyToOne(fetch = FetchType.EAGER)
    @JsonIgnoreProperties(value = {"sucursales"}, allowSetters = true)
    @JoinColumn(name = "id_ingrediente")
    private Ingrediente ingrediente;

    @ManyToOne(fetch = FetchType.EAGER)
    @JsonIgnoreProperties(value = {"sucursales"}, allowSetters = true)
    @JoinColumn(name = "id_articulo")
    private ArticuloVenta articuloVenta;

    @ManyToOne(fetch = FetchType.LAZY)
    @JsonIgnoreProperties(value = {"sucursal"}, allowSetters = true)
    @JoinColumn(name = "id_stock_entrante")
    private StockEntrante stockEntrante;

    @Column(name = "modificar_precio")
    private boolean modificarPrecio = false;


}
